sstables/sstable_set.cc

```diff
@@ -206,14 +206,19 @@ sstable_set::incremental_selector::~incremental_selector() = default;
 sstable_set::incremental_selector::incremental_selector(sstable_set::incremental_selector&&) noexcept = default;
 
 sstable_set::incremental_selector::selection
-sstable_set::incremental_selector::select(const dht::ring_position_view& pos) const {
-    if (!_current_range_view || !_current_range_view->contains(pos, _cmp)) {
-        std::tie(_current_range, _current_sstables, _current_next_position) = _impl->select(pos);
+sstable_set::incremental_selector::select(selector_pos s) const {
+    if (!_current_range_view || !_current_range_view->contains(s.pos, _cmp)) {
+        std::tie(_current_range, _current_sstables, _current_next_position) = _impl->select(s);
```

Comment:
ring_position_view is 24 bytes in size and the additional pointer to the range is 8 bytes long, so well be copying 32 bytes.  That seems a bit much.  Can we pass a reference to the selector_pos instead?
Those functions are synchronous, so it shouldn't fall out scope.

Reply:
Ok, you're asking for incremental_selector_impl::select() to receive 'const selector_pos&' instead? Yes, I can do that.

Reply:
yes, thanks

Reply:
thank you.